### PR TITLE
rename Domain.DomainName and Domain.TagList

### PR DESCRIPTION
**Pull request by jaypipes**
Renames Domain.DomainName to Domain.Name and Domain.TagList to
Domain.Tags to align the opensearchservice controller with other ACK
controllers and CRDs.

NOTE: This is a backwards-incompatible API change but I figure this
would be better to do early on in preview now than later...

Signed-off-by: Jay Pipes <jaypipes@gmail.com>

By submitting this pull request, I confirm that my contribution is made
under the terms of the Apache 2.0 license.

### CODE REVIEW

File: config/crd/common/bases/services.k8s.aws_adoptedresources.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@ apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: v0.7.0
+    controller-gen.kubebuilder.io/version: v0.4.0
```
Comment (RedbackThomson) [inline config/crd/common/bases/services.k8s.aws_adoptedresources.yaml:7]: How did this happen?

Comment (jaypipes) [reply config/crd/common/bases/services.k8s.aws_adoptedresources.yaml:7]: guess this is the version of controller-gen I have installed locally...

Comment (RedbackThomson) [reply config/crd/common/bases/services.k8s.aws_adoptedresources.yaml:7]: The build scripts should keep your kubebuilder inline with the version required. This seems like a mistake.

Comment (oleg-yudovich) [reply config/crd/common/bases/services.k8s.aws_adoptedresources.yaml:7]: Try to remove your local `bin` folder and run the `make` again

**Comment by RedbackThomson** [comment]
@jaypipes Needs to be regenerated with runtime `v0.17.0`, and then we can merge :)

**Pull request merged**